packages/core/core/src/requests/TargetRequest.js

```diff
@@ -135,64 +158,80 @@ export class TargetResolver {
+                optionTargets,
+                null,
+                '\t',
+              );
+              throw new ThrowableDiagnostic({
+                diagnostic: {
+                  message: `Missing distDir for target "${name}"`,
+                  origin: '@parcel/core',
+                  codeFrame: {
+                    code: optionTargetsString,
+                    codeHighlights: generateJSONCodeHighlights(
+                      optionTargetsString || '',
+                      [
+                        {
+                          key: `/${name}`,
+                          type: 'value',
+                        },
+                      ],
+                    ),
+                  },
                 },
-              },
-            });
-          }
-          let target: Target = {
-            name,
-            distDir: path.resolve(this.fs.cwd(), distDir),
-            publicUrl:
-              descriptor.publicUrl ??
-              this.options.defaultTargetOptions.publicUrl,
-            env: createEnvironment({
-              engines: descriptor.engines,
-              context: descriptor.context,
-              isLibrary: descriptor.isLibrary,
-              includeNodeModules: descriptor.includeNodeModules,
-              outputFormat: descriptor.outputFormat,
-              shouldOptimize:
-                this.options.defaultTargetOptions.shouldOptimize &&
-                descriptor.optimize !== false,
-              shouldScopeHoist:
-                this.options.defaultTargetOptions.shouldScopeHoist &&
-                descriptor.scopeHoist !== false,
-              sourceMap: normalizeSourceMap(this.options, descriptor.sourceMap),
-            }),
-          };
-
-          if (descriptor.distEntry != null) {
-            target.distEntry = descriptor.distEntry;
-          }
-
-          return target;
-        });
+              });
+            }
+            let target: Target = ({
+              name,
+              distDir: path.resolve(this.fs.cwd(), distDir),
+              publicUrl:
+                descriptor.publicUrl ??
+                this.options.defaultTargetOptions.publicUrl,
+              env: createEnvironment({
+                engines: descriptor.engines,
+                context: descriptor.context,
+                isLibrary: descriptor.isLibrary,
+                includeNodeModules: descriptor.includeNodeModules,
+                outputFormat: descriptor.outputFormat,
+                shouldOptimize:
+                  this.options.defaultTargetOptions.shouldOptimize &&
+                  descriptor.optimize !== false,
+                shouldScopeHoist:
+                  this.options.defaultTargetOptions.shouldScopeHoist &&
+                  descriptor.scopeHoist !== false,
+                sourceMap: normalizeSourceMap(
+                  this.options,
+                  descriptor.sourceMap,
+                ),
+              }),
+            }: Target);
```

Comment:
The cast on the right in `let target: Target = ({...}: Target)` isn't needed in any case

Reply:
it was needed (along with some other things) at one point in my flow type checker wrangling... but yes, it's not needed any longer. Removed.